cmd/counters/src/lib.rs

```diff
@@ -259,6 +311,342 @@ fn counter_dump(
     Ok(())
 }
 
+fn ipc_counter_dump(
+    hubris: &HubrisArchive,
+    core: &mut dyn Core,
+    subargs: &CountersArgs,
+) -> Result<()> {
+    let mut ipcs = BTreeMap::new();
+    for (varname, var) in hubris.qualified_variables() {
+        if varname.ends_with("_CLIENT_COUNTERS") {
+            let task = taskname(hubris, var)?;
+            if let Some(ref name) = subargs.name {
+                if !varname.contains(name) {
+                    continue;
+                }
+            }
+
+            let def = hubris.lookup_struct(var.goff)?;
+            let ctrs = load_counters(hubris, core, def, var)?;
+            let iface = ipcs.entry(varname).or_insert_with(|| {
+                let name = def
+                    .name
+                    .split("::")
+                    .last()
+                    .map(|s| {
+                        s.trim_end_matches("Counts").trim_end_matches("Event")
+                    })
+                    .unwrap_or("");
+                IpcIface { name, counters: Default::default() }
+            });
+            IpcCounters::populate(
+                &mut iface.counters,
```

Comment:
Nit: passing `&mut iface.counters` confused me here, so I think this would be more clear as a `populate_counters(&mut self, ..)` method on`IpcIface`.

Reply:
It's also called recursively by nested counters, so it can't be a method. I can try to make this a little bit clearer, though.

Reply:
@mkeeter what do you think of https://github.com/oxidecomputer/humility/pull/452/commits/fe56f03e991fb6d1fb8dc0a30a325a047bf5e78e? I've factored out the `IndexMap<String,  IpcCounters<'a>>` into a newtype with a `.populate()` method, so it's now only called on that (and the unreachable!() case is no longer possible). but, the code is a bit more complex now as there are a few more types...

Reply:
I like it, thanks!